api/app/src/routes/medical/patient.ts

```diff
@@ -74,22 +74,30 @@ router.put(
     const facilityId = getFromQueryOrFail("facilityId", req);
     const payload = patientUpdateSchema.parse(req.body);
 
+    const patient = await getPatientOrFail({ id, cxId });
+
+    if (patient.data.documentQueryStatus === "processing") {
+      return res.status(status.LOCKED).json("Document querying currently in progress");
+    }
+
```

Comment:
Can we please move this logic to the command, to leave the route focused on HTTP/conversion? 😃